Godeps/_workspace/src/github.com/cloudflare/cfssl/crypto/pkcs11key/pkcs11key.go

```diff
@@ -167,14 +167,25 @@ func (ps *PKCS11Key) Destroy() {
 func (ps *PKCS11Key) openSession() (session pkcs11.SessionHandle, err error) {
 	// Check if there is a PCKS11 token with slots. It has side-effects that
 	// allow the rest of the code here to work.
-	_, err = ps.module.GetSlotList(true)
+	slotIDs, err = ps.module.GetSlotList(true)
 	if err != nil {
 		return
 	}
 
-	slotID := uint(ps.slotID)
+	// Check that the desired slot exists
+	slotExists := false
+	for _, slotID := range slotIDs {
+		if slotID == ps.slotID {
+			slotExists = true
+			break
+		}
+	}
+	if !slotExists {
+		err = fmt.Errorf("Slot %u not found", ps.slotID)
+	}
+
 	// Look up slot by id
-	tokenInfo, err := ps.module.GetTokenInfo(slotID)
+	tokenInfo, err := ps.module.GetTokenInfo(ps.slotID)
```

Comment:
Given comments from @jsha in #784, it seems like we need to keep the `GetSlotList` entry.  I've added a comment that explains that we need this to accommodate some HSMs.  

I still think we should do the search through the slot list, since it produces a clearer error.